lib/spack/docs/index.rst

```diff
@@ -37,21 +37,24 @@ package:
 If you're new to spack and want to start using it, see :doc:`getting_started`,
 or refer to the full manual below.
 
------------------
-Table of Contents
------------------
-
 .. toctree::
    :maxdepth: 2
+   :caption: Documentation and Tutorial
 
    features
    getting_started
    basic_usage
+   workflows
+
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Reference Manual
+
    packaging_guide
    mirrors
    configuration
    developer_guide
-   case_studies
    command_index
    package_list
    API Docs <spack>
```

Comment:
We have two orthogonal ways we want to organize here:
 1. User / Developer
 2. Tutorial / Reference

I'm dubious of (1) because there are many grades of users and developers.
Who's the user and who's the developer?  I can think of:

 1. People who load modules and run things in the /bin directories of Spack
packages.
 2. Sysadmins who run Spack builds for Spack users.
 3. People who install software written by others, where installation
instructions tell them to run Spack.
 4. People who write software against the /lib and /include directories of
Spack packages --- using one of spack setup, module loads, filesystem
views.  (If they use spack setup, the most seamless and lowest fuss
approach, they will be writing Spack packages for their stuff).
 5. People who add packages to Spack for software others wrote.
 6. People who add packages to Spack for software they wrote.
 7. People who hack the core Spack python code because they found bugs.
 8. People who hack core Spack to add new features.
 9. People who edit Spack documentation but can't (or aren't at the moment)
modifying Python code.

My conclusion is we should probably NOT try to organize the manual based on
users vs. developers.  Because I can't tell who's who.


On Fri, Sep 2, 2016 at 1:16 PM, Adam J. Stewart <notifications@github.com>
wrote:

> In lib/spack/docs/index.rst:
>
> >     packaging_guide
> >     mirrors
> >     configuration
> >     developer_guide
> > -   case_studies
> >     command_index
> >     package_list
> >     API Docs <spack>
>
> I've been thinking about this. How do you feel about the following
> structure:
>
> .. toctree::
>    :maxdepth: 2
>    :caption: User Documentation
>
>    features
>    getting_started
>    basic_usage
>    workflows
>    configuration
>    mirrors
>    package_list
>    command_index
>
> .. toctree::
>    :maxdepth: 2
>    :caption: Developer Documentation
>
>    contribution_guide  # see note below
>    packaging_guide
>    developer_guide
>    API Docs <spack>
>
> @davydden <https://github.com/davydden> is adding a Contribution Guide in
> #1664 <https://github.com/LLNL/spack/pull/1664>. Right now, it mostly
> contains git shortcuts, but I plan on adding documentation about how to run
> all of the tests and how you need the tests to pass in order to get a PR
> merged.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1676/files/97cd009600fbdb2bbc654f6c0bdb1813436ef4c7#r77381158>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cd10ppCEmBuqqgS6d5uQuSp4fHRa3ks5qmFn7gaJpZM4JxOaa>
> .
>